node/actors/bft/src/leader/replica_commit.rs

```diff
@@ -134,16 +112,22 @@ impl StateMachine {
             .insert(author.clone(), signed_message);
 
         // Now we check if we have enough messages to continue.
-        let num_messages = self.commit_message_cache.get(&message.view).unwrap().len();
-
-        if num_messages < consensus.threshold() {
-            return Err(Error::NumReceivedBelowThreshold {
-                num_messages,
-                threshold: consensus.threshold(),
-            });
+        let mut by_proposal: HashMap<_, Vec<_>> = HashMap::new();
+        for msg in self
+            .commit_message_cache
+            .get(&message.view)
+            .unwrap()
+            .values()
```

Comment:
Nit: IIUC, this could be simplified by extracting an updated cache entry into a var:

```rust
let cache_entry = self.commit_message_cache
    .entry(message.view)
    .or_default();
cache_entry.insert(author.clone(), signed_message);

// Now we check if we have enough messages to continue.
let mut by_proposal: HashMap<_, Vec<_>> = HashMap::new();
for msg in cache_entry.values() {
    by_proposal.entry(msg.msg.proposal).or_default().push(msg);
}
```